### DISCUSSION TITLE
Importing glTF file from byte array

### DISCUSSION THREAD

**Discussion by Zamaroth** [atteneder/glTFast#395] [Q&A] [answered]
Hello, I want to get a file from server via request and then import it into the scene. I know how to do it if the file is in glb format([this](https://github.com/atteneder/glTFast/blob/main/Documentation~/ImportRuntime.md#load-from-byte-array) but changing it to request).  

However I cannot figure out how to load a glTF file the same way. The glTF looks like this: 
![image](https://user-images.githubusercontent.com/19417407/166957189-2c9c0225-672b-44ee-a4b2-01a28598a509.png)
Do I need to create a custom downloader to also make requests for the bin and png file?  

The documentation says: `Note: Most users want to load self-contained glTF binary files this way, but LoadGltfBinary also takes the original URI of glTF file as second parameter, so it can resolve relative URIs.`  
What is `m_path` in the example? I tried adding the url of gltf as `m_path`, but it doesn't work and LoadGltfBinary just returns false.

**Comment by atteneder** [comment, marked as answer]
Hi,
Granted, that example code is not clear at all.

It's meant like so:


```csharp
async void LoadGltfBinaryFromMemory() {
    var filePath = "/path/to/file.glb";
    byte[] data = File.ReadAllBytes(filePath);
    var gltf = new GltfImport();
    bool success = await gltf.LoadGltfBinary(data, new Uri(filePath));
    if (success) {
        success = gltf.InstantiateMainScene(transform);
    }
}
```

So the URI provided is the one of the `.gltf` file. It'll be used as reference for relative URIs within the JSON.

hth

**Reply by Zamaroth** [reply]
I tried it, but the glTF wouldn't load... the LoadGltfBinary method just returns false immediately without any error.  

I made a git repo here: [https://github.com/Zamaroth/glTFastImportTest](https://github.com/Zamaroth/glTFastImportTest)  
It contains basically just your code with the path changed to a included glTF model.  

The data is not null and it fails somewhere inside LoadGltfBinary method, as it gets there... Does the filePath needs to be relative, or what am I doing wrong?

**Reply by atteneder** [reply]
Sorry about that. I'll give it a try soon.

**Reply by atteneder** [reply]
Ok, I had a look

First, `LoadGltfBinary` only loads glTF-binary (as its name suggests and its documentation states). I could make a similar override that loads a JSON glTF from buffer or string, but can you tell me in which use-case this would be benefitial over GltfImport.Load(URI) ?

Second, the actual error was not logged onto the console, since you did not set a default logger. I'm starting to think using the `ConsoleLogger` as default fallback (and let users override it to actually null) is user-friendlier. You would have found out right away.

**Reply by Zamaroth** [reply]
Oh my bad... for some reason I thought I have to handle requests from server by myself and .Load is solely for loading from files system.  
Now I see, you can just put url into .Load and it works for both glb and glTF. And if I need to add any headers, I can create custom DownloadProvider.  

Anyway, thank you for you help in helping me understand this better :)
